tests/test_dataset_dict.py

```diff
@@ -4,21 +4,147 @@
+        for dset_split in dset.values():
+            with self.assertRaises(TypeError):
+                dset_split[0]
+
+    @require_tf
+    def test_set_format_tf(self):
+        import tensorflow as tf
+
+        dset = self._create_dummy_dataset_dict(multiple_columns=True)
+        dset.set_format(type="tensorflow", columns=["col_1"])
+        for dset_split in dset.values():
+            self.assertEqual(len(dset_split[0]), 1)
+            self.assertIsInstance(dset_split[0]["col_1"], tf.Tensor)
+            self.assertListEqual(list(dset_split[0]["col_1"].shape), [])
+            self.assertEqual(dset_split[0]["col_1"].numpy().item(), 3)
+
+        dset.set_format(type="tensorflow", columns=["col_1"], output_all_columns=True)
+        for dset_split in dset.values():
+            self.assertEqual(len(dset_split[0]), 2)
+            self.assertIsInstance(dset_split[0]["col_2"], str)
+            self.assertEqual(dset_split[0]["col_2"], "a")
+
+        dset.set_format(type="tensorflow", columns=["col_1", "col_2"])
+        for dset_split in dset.values():
+            self.assertEqual(len(dset_split[0]), 2)
+            self.assertEqual(dset_split[0]["col_2"].numpy().decode("utf-8"), "a")
+
+    def test_set_format_pandas(self):
+        dset = self._create_dummy_dataset_dict(multiple_columns=True)
+        dset.set_format(type="pandas", columns=["col_1"])
+        for dset_split in dset.values():
+            self.assertEqual(len(dset_split[0].columns), 1)
+            self.assertIsInstance(dset_split[0], pd.DataFrame)
+            self.assertListEqual(list(dset_split[0].shape), [1, 1])
+            self.assertEqual(dset_split[0]["col_1"].item(), 3)
+
+        dset.set_format(type="pandas", columns=["col_1", "col_2"])
+        for dset_split in dset.values():
+            self.assertEqual(len(dset_split[0].columns), 2)
+            self.assertEqual(dset_split[0]["col_2"].item(), "a")
+
+    def test_set_format(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            dsets = self._create_dummy_dataset_dict()
+
+            mapped_dsets_1: Dict[str, Dataset] = dsets.map(
+                lambda ex: {"foo": ["bar"] * len(ex["filename"])}, batched=True
+            )
+            self.assertListEqual(list(dsets.keys()), list(mapped_dsets_1.keys()))
+            self.assertListEqual(mapped_dsets_1["train"].column_names, ["filename", "foo"])
+
+            cache_file_names = {
+                "train": os.path.join(tmp_dir, "train.arrow"),
+                "test": os.path.join(tmp_dir, "test.arrow"),
+            }
+            mapped_dsets_2: Dict[str, Dataset] = mapped_dsets_1.map(
+                lambda ex: {"bar": ["foo"] * len(ex["filename"])}, batched=True, cache_file_names=cache_file_names
+            )
+            self.assertListEqual(list(dsets.keys()), list(mapped_dsets_2.keys()))
+            self.assertListEqual(mapped_dsets_2["train"].column_names, ["filename", "foo", "bar"])
```

Comment:
This test doesn't seem to test `set_format`

Reply:
Oh yes!